src/compiler/emitter.ts

```diff
@@ -924,6 +930,14 @@ export function emitFiles(resolver: EmitResolver, host: EmitHost, targetSourceFi
         forEachChild(node, collectLinkedAliases);
     }
 
+    function markLinkedReferences(file: SourceFile) {
+        ts.forEachChildRecursively(file, n => {

```

Comment:
can this instead check if file doesnt have `TypeChecked` flag and do this work and then we can get rid of `getDiagnostics` all together from `getEmitResolver()`

Reply:
Not quite - the `TypeChecked` flag is, ironically, set on files whose type checking is skipped. Maybe it's a misnomer, but it really just means "Has `checkSourceFile` been called on this node?". I get your drift though and, yeah, we could probably do a bit of refactoring to use a node flag to track the diagnostic walk state of the containing file. I'll probably put that up as a separate cleanup, though. I also think I want to move around/refactor the API logic for `collectLinkedAliases` so the API is harder to misuse, and will probably change `markLinkedReferences` to match, so I'll put all those nonfunctional changes in a separate PR (which should have a 0-test diff).